File: Java/src/main/java/org/onlineoptimisation/problem/benchmark/EllipsoidalFunctionRotated.java
package org.onlineoptimisation.problem.benchmark;

import org.armadillojava.Arma;
import org.armadillojava.Col;
import org.armadillojava.Mat;

public class EllipsoidalFunctionRotated extends BenchmarkProblem {

  protected Mat _R;
  
  public EllipsoidalFunctionRotated(int numberOfDimensions) {
    super(numberOfDimensions);
    
    _R = randomRotation();
  }

  @Override
  public double getObjectiveValueImplementation(Col parameter) {
    return Arma.dot(scaling(1000000), Arma.square(oscillate(_R.times(parameter))));
  }

}

File: src/optimisationProblem/benchmark/ellipsoidalFunctionRotated.cpp
#include <hop_bits/optimisationProblem/benchmark/ellipsoidalFunctionRotated.hpp>

namespace hop {
  double EllipsoidalFunctionRotated::getObjectiveValueImplementation(const arma::Col<double>& parameter) const {
    return arma::dot(scaling_, arma::square(getOscillationTransformation(rotationR_ * (parameter - translation_))));
  }

  std::string EllipsoidalFunctionRotated::to_string() const {
    return "EllipsoidalFunctionRotated";
  }
}
